src/util/UnixSignalHandler.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <QtGlobal>
+#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
+#    include <QtSystemDetection>
+#endif
+
+#if defined(Q_OS_UNIX)
+#    include <QObject>
+#    include <QSocketNotifier>
+
+#    include <array>
+
+namespace chatterino {
+
+class UnixSignalHandler : public QObject
+{
+    Q_OBJECT
+
+public:
+    UnixSignalHandler(int signal);
+
+    static void fired(int signal);
+
+Q_SIGNALS:
+    void signalFired();
+
+private Q_SLOTS:
```

Comment:
You don't really _need_ `Q_SLOTS` here, since you pass a function pointer to `handleSignal` instead of a name.